src/db_options.rs

```diff
@@ -127,6 +127,13 @@ impl Env {
         }
     }
 
+    /// Returns a new environment which wraps and takes ownership of the provided
+    /// raw environment.
+    // Marked unsafe because ownership is transferred to the returned Env.
+    pub unsafe fn from_raw(env: *mut ffi::rocksdb_env_t) -> Self {
```

Comment:
Couple of questions.
1. Why ?
2. What is gonna be a caller of this method ?
3. Where are tests for this method ?
4. And we shouldn't forget about such [case](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e5feb2a24535644c715cadf18651fd93) ?

Reply:
1) The rocks environment provides mechanisms to hook into lower-level subsystems of Rocks, such as file IO. For example the [rocks gazette consumer store](https://gazette.readthedocs.io/en/latest/consumers-concepts.html#stores) leverages this to replicate databases states as they're being written. You're deep into FFI land to do this type of thing, of course, but once you've _built_ an instrumented FFI environment it's nice to hand it to this crate for usage.

2) [Example call site](https://github.com/estuary/flow/blob/master/crates/derive/src/derive_api.rs#L76-L80). You have to already have an FFI *env pointer of course. This PR merely plumbs through a means to hand it to an Env of this crate, and as best I can tell matches the standard conventions for exposing and naming an unsafe operation of this kind in the standard library and elsewhere.

3) I'm not sure what it would cover, that the compiler doesn't check already. If there were a paired `into_raw()` function then a test case might exercise round-tripping through that raw pointer, but I don't get the feeling an `into_raw` is desired and I'm not suggesting it. (FWIW there's of course plenty of integrated, end-to-end coverage in our repo [that uses this API](https://github.com/estuary/flow)).

4) That's why this function is unsafe, and specifically notes it's unsafe because ownership is transferred, similarly to how `Vec::from_raw_parts` and other functions of this kind are documented.